base_layer/wallet/src/transaction_service/storage/sqlite_db.rs

```diff
@@ -265,6 +266,26 @@ impl TransactionBackend for TransactionServiceSqliteDatabase {
             CompletedTransactionSql::find(tx_id, &(*conn)).is_ok())
     }
 
+    fn get_pending_transaction_counterparty_pub_key_by_tx_id(
+        &self,
+        tx_id: u64,
+    ) -> Result<CommsPublicKey, TransactionStorageError>
+    {
+        let conn = acquire_lock!(self.database_connection);
+        if let Ok(outbound_tx_sql) = OutboundTransactionSql::find(tx_id, &(*conn)) {
+            if let Ok(outbound_tx) = OutboundTransaction::try_from(outbound_tx_sql) {
```

Comment:
Shouldn't this return an error if it finds the transaction by ID but cannot convert it, rather than querying inbound transactions (which will fail) and returning a ValueNotFound error?

Reply:
Ja, I suppose that is more correct. In the end if this conversion fails the db is corrupt but I suppose should def throw and error in that case.